desktop/core/src/desktop/js/apps/admin/Overview/Analytics.tsx

```diff
@@ -14,63 +14,82 @@
 import { i18nReact } from '../../../utils/i18nReact';
 import Input from 'cuix/dist/components/Input';
-import { post } from '../../../api/utils';
-import { ANALYTICS_PREFERENCES_API_URL } from '../Components/utils';
+import { GET_USAGE_ANALYTICS_API_URL, UPDATE_USAGE_ANALYTICS_API_URL } from '../Components/utils';
+import { HueAlert } from '../../../reactComponents/GlobalAlert/types';
+import { GLOBAL_INFO_TOPIC } from '../../../reactComponents/GlobalAlert/events';
+import useLoadData from '../../../utils/hooks/useLoadData/useLoadData';
+import useSaveData from '../../../utils/hooks/useSaveData/useSaveData';
+import LoadingErrorWrapper from '../../../reactComponents/LoadingErrorWrapper/LoadingErrorWrapper';
 import './Overview.scss';
 
-interface UpdatePreferences {
-  status: number;
+interface UsageAnalyticsResponse {
+  analyticsEnabled?: boolean;
   message?: string;
 }
 
 const Analytics = (): JSX.Element => {
-  const [collectUsage, setCollectUsage] = useState<boolean>(false);
   const { t } = i18nReact.useTranslation();
+  const {
+    data: usageAnalyticsData,
+    loading: loadingAnalytics,
+    error: usageAnalyticsError,
+    reloadData
+  } = useLoadData<UsageAnalyticsResponse>(GET_USAGE_ANALYTICS_API_URL);
 
-  const saveCollectUsagePreference = async (collectUsage: boolean) => {
-    const response = await post<UpdatePreferences>(ANALYTICS_PREFERENCES_API_URL, {
-      collect_usage: collectUsage
-    });
-
-    if (response.status === 0) {
-      const successMessage = collectUsage
+  const {
+    save: updateAnalyticsPreference,
+    loading: updatingAnalyticsPreference,
+    error: updateAnalyticsPreferenceError
+  } = useSaveData<{ analytics_enabled: boolean }>(UPDATE_USAGE_ANALYTICS_API_URL, {
+    onSuccess: response => {
+      reloadData();
+      const successMessage = response.analytics_enabled
         ? t('Analytics have been activated.')
         : t('Analytics have been deactivated.');
-      huePubSub.publish('hue.global.info', { message: successMessage });
-    } else {
-      const errorMessage = collectUsage
-        ? t('Failed to activate analytics.')
-        : t('Failed to deactivate analytics.');
-      huePubSub.publish('hue.global.error', { message: errorMessage });
+      huePubSub.publish<HueAlert>(GLOBAL_INFO_TOPIC, { message: successMessage });
     }
-  };
+  });
+
+  const errors = [
+    {
+      enabled: !!usageAnalyticsError,
+      message: usageAnalyticsError ?? t('An unknown error occurred while fetching data.')
+    },
+    {
+      enabled: !!updateAnalyticsPreferenceError,
+      message: updateAnalyticsPreferenceError ?? t('Failed to update analytics.')
+    }
+  ];
 
-  const handleCheckboxChange = event => {
-    const newPreference = event.target.checked;
-    setCollectUsage(newPreference);
-    saveCollectUsagePreference(newPreference);
+  const handleAnalyticsCheckboxChange = (checked: boolean) => {
+    const formData = new FormData();
```

Comment:
If we are using the new public APIs why are we still sending formData and not json?

@Harshg999 @ramprasadagarwal

Reply:
@bjornalm - Changed as part of https://github.com/cloudera/hue/pull/4171